File: python/kms-generate-data-key.py
import sys
from base64 import b64encode

import boto3

client = boto3.client('kms')

if len(sys.argv) != 2:
    print('Usage: {} <keyId|alias>'.format(sys.argv[0]))
    sys.exit(1)

response = client.generate_data_key_without_plaintext(
    KeyId=sys.argv[1],
    KeySpec='AES_256',
    GrantTokens=[
        'string'
    ]
)

data_key = b64encode(response["CiphertextBlob"]).decode('utf-8')
print(data_key)

File: node/kmsGenerateDataKey.js
const path = require('path');

const { argv } = process;
const { KMSClient, GenerateDataKeyCommand } = require('@aws-sdk/client-kms');

const client = new KMSClient();

if (argv.length !== 3) {
  const [node, script] = argv.slice(0, 2).map((p) => path.parse(p).base);
  console.log(`Usage: ${node} ${script} <kmsKeyId|alias>`);
  process.exit(1);
}

client
  .send(
    new GenerateDataKeyCommand({
      KeyId: argv[2],
      KeySpec: 'AES_256',
    })
  )
  .then((result) => {
    const dataKey = Buffer.from(result.CiphertextBlob).toString('base64');
    console.log(dataKey);
  });
